contributors/design-proposals/auditing.md

```diff
@@ -0,0 +1,276 @@
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+**WORK-IN-PROCESS -- UNFINISHED -- WORK-IN-PROGRESS**
+
+## Abstract
+
+This proposal aims at extending the auditing log capabilities of the apiserver.
+
+## Motivation and Goals
+
+With https://github.com/kubernetes/kubernetes/pull/27087 basic audit logging is added to Kubernetes. It basically implements `access.log` like http handler based logging of all requests in the apiserver API. It does not do deeper inspection of the API calls or of their payloads. Moreover, it has no specific knowledge of the API objects which are modified. Hence, the log output does not answer the question how API objects actually change.
+
+The log output format of https://github.com/kubernetes/kubernetes/pull/27087 is fixed. It is text based, unstructured (e.g. non-JSON) data which must be parsed to be usable in any advanced external system used to analyze audit logs.
+
+The log output format does not follow any public standard like e.g. https://www.dmtf.org/standards/cadf.
+
+With this proposal we describe how the auditing functionality can be extended in order:
+
+- to allow multiple output formats, e.g. access.log style or structured JSON output
+- to allow deep payload inspection to allow
+  - either real differential JSON output (which field of an object have changed from which value to which value)
+  - or full object output of the new state (and optionally the old state)
+- to be extensible in the future to fully comply with the Cloud Auditing Data Federation standard (https://www.dmtf.org/standards/cadf)
+- to allow filtering of the output
+  - by kind, e.g. don't log endpoint objects
+  - by object path (JSON path), e.g. to ignore all `*.status` changes
+  - by user, e.g. to only log end user action, not those of the controller-manager and scheduler
+  - by level (request headers, request object, storage object)
+
+while
+
+- not degrading apiserver performance when auditing is disabled.
+
+## Constraints and Assumptions
+
+* it is not the goal to implement all output formats one can imagine. The  main goal is to be extensible with a clear golang interface. Implementations of e.g. CADF must be possible, but won't be discussed here.
+* dynamic loading of backends for new output formats are out of scope.
+
+## Use Cases
+
+1. As a cluster operator I want to enable audit logging of requests to the apiserver in order **to comply with given business regulations** regarding a subset of the 7 Ws of auditing:
+
+  - **what** happened?
+  - **when** did it happen?
+  - **who** initiated it?
+  - **on what** did it happen (e.g. pod foo/bar)?
+  - **where** was it observed (e.g. apiserver hostname)?
```

Comment:
The 7 W questions are coming [CADF](https://www.dmtf.org/sites/default/files/standards/documents/DSP2038_1.1.0.pdf) but I can improve the description.